### DISCUSSION TITLE
How can I use my custom Ollama model in this project?

### DISCUSSION THREAD

**Discussion by vatche-t** [BrainBlend-AI/atomic-agents#41] [Q&A] [answered]
Hello!
I'm interested in integrating my custom Ollama model into this project. Could you please guide me on:

    How to configure the project to work with a locally hosted model like Ollama?
    Any specific requirements or dependencies I should be aware of?
    Example setups or configurations for similar integrations would be very helpful.

Thank you in advance!

**Comment by cediorio** [comment, marked as answer]
I was able to use ollama easily. Here's the relevant code:

```
import instructor
from openai import OpenAI

# ...

client = instructor.from_openai(
    OpenAI(
        base_url="http://localhost:11434/v1",
        api_key="ollama", # not actually used...
    ),
    mode=instructor.Mode.JSON
)
```

**Comment by YuyangXueEd** [comment]
Currently my practice is to follow Instructor's doc:
```
client = instructor.from_provider("ollama/qwen3", async_client=False, mode=instructor.Mode.JSON)
model = "qwen3"
```
